examples/2-challenges/water2_gaussian_extreme_criteria/saved/2023-01-10/water2.log

```diff
@@ -0,0 +1,172 @@
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   29 : Displace = [0m1.983e-07[0m/[0m2.977e-07[0m (rms/max) Trust = 1.414e-04 ([92m+[0m) Grad = [0m8.686e-10[0m/[0m1.508e-09[0m (rms/max) E (change) = -149.9412443593 ([92m-1.137e-13[0m) Quality = [0m204.869[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   30 : Displace = [0m1.951e-07[0m/[0m2.980e-07[0m (rms/max) Trust = 2.000e-04 ([92m+[0m) Grad = [0m2.760e-10[0m/[0m2.927e-10[0m (rms/max) E (change) = -149.9412443593 ([92m-1.990e-13[0m) Quality = [0m421.922[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   31 : Displace = [0m1.066e-08[0m/[0m1.736e-08[0m (rms/max) Trust = 2.828e-04 ([92m+[0m) Grad = [92m5.254e-11[0m/[92m8.832e-11[0m (rms/max) E (change) = -149.9412443593 ([92m+5.116e-13[0m) Quality = [91m-76283.503[0m
+[93mNot rejecting step - RMS displacement close to tmin = 1.000e-04[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   32 : Displace = [92m2.415e-09[0m/[92m3.351e-09[0m (rms/max) Trust = 1.000e-04 ([91m-[0m) Grad = [0m6.998e-10[0m/[0m1.003e-09[0m (rms/max) E (change) = -149.9412443593 ([92m-3.979e-13[0m) Quality = [0m2935064.631[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   33 : Displace = [0m4.324e-07[0m/[0m6.907e-07[0m (rms/max) Trust = 1.414e-04 ([92m+[0m) Grad = [0m1.063e-09[0m/[0m1.416e-09[0m (rms/max) E (change) = -149.9412443593 ([92m+1.990e-13[0m) Quality = [91m-150.019[0m
+[93mNot rejecting step - RMS displacement close to tmin = 1.000e-04[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   34 : Displace = [0m4.479e-07[0m/[0m7.162e-07[0m (rms/max) Trust = 1.000e-04 ([91m-[0m) Grad = [0m9.049e-10[0m/[0m1.078e-09[0m (rms/max) E (change) = -149.9412443593 ([92m+3.126e-13[0m) Quality = [91m-221.895[0m
+[93mNot rejecting step - trust below tmin = 1.000e-04[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   35 : Displace = [0m2.274e-08[0m/[0m3.340e-08[0m (rms/max) Trust = 1.000e-04 (=) Grad = [0m1.337e-10[0m/[0m2.394e-10[0m (rms/max) E (change) = -149.9412443593 ([92m-4.263e-13[0m) Quality = [0m24148.848[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
+Step   36 : Displace = [92m5.386e-09[0m/[92m8.318e-09[0m (rms/max) Trust = 1.414e-04 ([92m+[0m) Grad = [92m2.850e-11[0m/[92m3.749e-11[0m (rms/max) E (change) = -149.9412443593 ([92m-1.990e-13[0m) Quality = [0m398433.022[0m
+Hessian Eigenvalues: 1.19973e-03 5.20439e-03 8.95998e-03 ... 5.66922e-01 5.84660e-01 7.04358e-01
```

Comment:
I notice that all of the energies reported from Gaussian here are the same to 10 dp despite the XYZ coords reported in the `water2_optim.xyz` being different. I also notice the energy change between steps once the energy converges seems to change between a few repeating values randomly. Iterations 25, 26, 28, 30 and 36 are `-1.990e-13` we also see iterations 27 and 31 have the same value which due to them being so small seem like noise from converging the SCF. Could it be that Gaussian only supports reading input coordinates to a certain precision or is it something else?

Reply:
Thanks, Josh - I think this is probably due to a precision limitation due to how Gaussian writes out the energy, or how the energy is stored as a double precision number.  In the `.fchk` file, the energy data looks like this: `-1.499412443593249E+02` - the smallest digit is on the order of 1e-13.  I think it's not printing out any more because the energy is stored as a double, which inherently has about 15-16 significant digits.

If the problem were solely due to the finite precision of the Gaussian output, I would expect energy changes to look like `1.000e-13, 2.000e-13` or something like that.  However, we're seeing energy differences that look like `-1.990e-13`.  I'm guessing this is because Python is storing the energy as a `float64`, which means when we take differences between such large numbers, the precision of the result is limited to only 1-2 bits in the double precision representation.  That could explain why the energy differences look somewhat "granular" in these final steps.

I agree the behavior seems odd but I don't think it's a solvable issue or a major issue.